capstone/src/main/java/com/step902020/capstone/IndividualController.java

```diff
@@ -0,0 +1,85 @@
+
+package com.step902020.capstone;
+
+import java.io.IOException;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+import com.google.gson.Gson;
+import java.io.IOException;
+
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@RestController
+public class IndividualController {
+
+  @Autowired
+  private IndividualRepository individualRepository;
+
+  @GetMapping("get-individual")
+  public List<Individual> getIndividual(@RequestParam("email") String email) {
+    return this.individualRepository.findByEmail(email);
+  }
+
+
+  @PostMapping("save-individual")
+  public RedirectView saveIndividual(
+      @RequestParam("firstname") String firstname,
+      @RequestParam("lastname") String lastname, 
+      @RequestParam("email") String email, 
+      @RequestParam("user-type") String userType,
+      @RequestParam("university") String university) throws IOException {
+
+    Individual current = null;
+    List<Individual> userList = this.individualRepository.findByEmail(email);
```

Comment:
I missed this in the previous PR. If `findByEmail()` is always expected to return at most one result, you can return `Optional<Individual>` instead of a list. It will simplify the logic for retrieving the current user in a few places.
https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repository-query-return-types

Reply:
If I do not find a user and it returns an optional.empty() how do I detect this on the front end when it is turned into a json? I couldn't find documentation for this.

Reply:
Ah, great question! It might require some experimentation. I think I would return `optionalResult.orElse(null)`, hoping that null will get serialized as an empty Json object. So the REST controller return type would just be `Individual`.
https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#orElse-T-

I had to google around for how Jackson (which Spring Web uses to serialize) deals with Optional types: https://www.baeldung.com/jackson-optional

Reply:
Sorry Elena, I think you might have seen the previous version of my comment. I was trying to see what value optional.empty() would return on the frontend but I edited the comment after I realized the Java version is incorrect on cloud shell. I don't know if I should go ahead and update Java.